src/wallet/spend.cpp

```diff
@@ -971,7 +973,26 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
     for (const auto& coin : selected_coins) {
         txNew.vin.push_back(CTxIn(coin->outpoint, CScript(), nSequence));
     }
-    DiscourageFeeSniping(txNew, rng_fast, wallet.chain(), wallet.GetLastBlockHash(), wallet.GetLastBlockHeight());
+
+    // `DiscourageFeeSniping` may backdate locktime to improve privacy of
+    // transactions which are delayed after signing for specific reasons (see
+    // comments in `DiscourageFeeSniping`), but we prohibit this for RBF
+    // replacement transactions and transactions spending unconfirmed
+    // outputs. The reason for this is that if the locktime of the transaction
+    // gets backdated below the locktime of the transaction it replaces or below
+    // the locktime of its unconfirmed parent, the "signed earlier but only
+    // broadcast now" explanation will become unrealistic, and this locktime
+    // discrepancy may serve as a fingerprint of the wallet being used.
+    bool locktime_backdate_allowed = true;
+    for (const auto& coin: selected_coins) {
+        const CWalletTx* coin_wtx{wallet.GetWalletTx(coin->outpoint.hash)};
+        if (wallet.IsSpent(coin->outpoint) || (coin_wtx && !wallet.IsTxInMainChain(*coin_wtx))) {
```

Comment:
In e979f2ebc8a74474d2bdfee61fd64472e4c4b873 "wallet: do not backdate locktime if it may help wallet fingerprinting"

Each `coin` already has a `depth` member which will be 0 if it is an unconfirmed coin. We can use this instead of having to lookup the parent `CWalletTx`.

Reply:
Thanks, that's a great simplification! Done.